api/app/controllers/spree/api/checkouts_controller.rb

```diff
@@ -34,8 +34,13 @@ def update
           end
 
           return if after_update_attributes
-          state_callback(:after) if @order.next
-          respond_with(@order, default_template: 'spree/api/orders/show')
+
+          if @order.next || @order.completed?  
+            state_callback(:after)
+            respond_with(@order, default_template: 'spree/api/orders/show')
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.